### PR TITLE
Expand Techdocs not found error handling

### PR DESCRIPTION
**Pull request by Crevil**
Currently if a document is requested that is found found a generic log line
containing only the error from the provider is logged.

    {"type":"plugin","plugin":"techdocs","message":"The specified key does not exist.","level":"warn","service":"backstage"}

This gives very little indication as to what and where something failed.

This change expands the log lines to include more details about what is being
attempted. It also removes the propagation of the raw error message to the
client as there might be information here that is not meant to be disclosed.

Tests are added for each provider. A bug in the Azure Blob storage provider was
found where the download method would reject its promise but the express handler
would not detect this and therefore not write a response.

Signed-off-by: Crevil <bjoern.soerensen@gmail.com>

Let me know if anything should be extracted and shared across the implementations.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/techdocs-common/src/stages/publish/azureBlobStorage.ts
Comment (iamEAP) [inline packages/techdocs-common/src/stages/publish/azureBlobStorage.ts:342]: That this is not standard now is probably a result of the community-driven nature of publisher development, but it would be good to standardize on a log-level for all of these, while we're here.

I think the intention in the past has been to keep these as `warn`s because they are notable, but not a major failure state of the app.

**Review by iamEAP — COMMENTED** [review]
Thanks for the contrib @Crevil!  See one comment inline, but otherwise looks good.

While you're updating: if you rename the changeset to be `techdocs-` prefixed, we can avoid waiting on the review by core reviewers too.

File: packages/techdocs-common/src/stages/publish/azureBlobStorage.ts
Comment (Crevil) [reply packages/techdocs-common/src/stages/publish/azureBlobStorage.ts:342]: Got it. Didn't actually notice that I had mixed the levels. Great catch 🙏

**Review by iamEAP — APPROVED** [review]
Thanks agains @Crevil!

**Pull request merged**
